vertices_to_h5m/core.py

```diff
@@ -151,24 +156,228 @@ def vertices_to_h5m(
+            material_tags=material_tags,
+            h5m_filename=h5m_filename,
+        )
+    else:
+        raise ValueError(f"method must be either pymoab or h5py, not {method}")
+
+
+def vertices_to_h5m_h5py(
+    vertices: Union[
+        Iterable[Tuple[float, float, float]], Iterable["cadquery.occ_impl.geom.Vector"]
+    ],
+    triangle_groups: Iterable[Tuple[int, int, int]],
+    material_tags: Iterable[str],
+    h5m_filename="dagmc.h5m",
+):
+    import h5py
+
+    vertices_floats = check_vertices(vertices)
+
+    local_triangle_groups = fix_normals(
+        vertices=vertices_floats, triangles_in_each_volume=triangle_groups
+    )
+
+    f = h5py.File(h5m_filename, "w")
+
+    all_triangles = np.vstack(local_triangle_groups)
+
+    # give each local group a unique tag
+    # tag_data = np.concatenate(
+    #     [np.full(len(group), i) for i, group in enumerate(local_triangle_groups)]
+    # )
+    # Alternative: Set all tags to -1
+    tag_data = np.full(len(all_triangles), -1)
+
+    tstt = f.create_group("tstt")
+
+    global_id = 1  # counts both triangles and coordinates
+    mesh_name = 2
+
+    nodes_group = tstt.create_group("nodes")
+    coords = nodes_group.create_dataset("coordinates", data=vertices)
+    coords.attrs.create("start_id", global_id)
+    global_id += len(vertices)
+
+    # tag all nodes with -1
+    node_tag_data = np.full(len(vertices), -1)
+    node_tags = nodes_group.create_group("tags")
+    node_tags.create_dataset("GLOBAL_ID", data=node_tag_data)
+
+    elements = tstt.create_group("elements")
+
+    elems = {
+        "Edge": 1,
+        "Tri": 2,
+        "Quad": 3,
+        "Polygon": 4,
+        "Tet": 5,
+        "Pyramid": 6,
+        "Prism": 7,
+        "Knife": 8,
+        "Hex": 9,
+        "Polyhedron": 10,
+    }
+    tstt["elemtypes"] = h5py.enum_dtype(elems)
+
+    now = datetime.now()
+    tstt.create_dataset(
+        "history",
+        data=[
+            "vertices_to_h5m",
+            __version__,
+            now.strftime("%m/%d/%y"),
+            now.strftime("%H:%M:%S"),
+        ],
+    )
+
+    tri3_group = elements.create_group("Tri3")
+    tri3_group.attrs.create("element_type", elems["Tri"], dtype=tstt["elemtypes"])
+
+    connectivity_group = tri3_group.create_dataset(
+        "connectivity",
+        data=all_triangles + 1,  # node indices are 1 based in h5m
+        dtype=np.uint64,
+    )
+
+    connectivity_group.attrs.create("start_id", global_id)
+    global_id += len(all_triangles)
 
+    # give each local group a unique tag
+    tag_data = np.concatenate(
+        [np.full(len(group), i) for i, group in enumerate(local_triangle_groups)]
+    )
+    tags_tri3_group = tri3_group.create_group("tags")
+    tags_tri3_group.create_dataset("GLOBAL_ID", data=tag_data)
+
+    tags_tstt_group = tstt.create_group("tags")
+
+    cat_group = tags_tstt_group.create_group("CATEGORY")
+    cat_group.attrs.create("class", 1, dtype=np.int32)
+    id_list = [7, 8, 9]
```

Comment:
this list looks hard coded for 3 volumes, TODO make this dynamic based on the input args